src/types.rs

```diff
@@ -399,11 +391,22 @@ impl From<ReturnCode> for u8 {
     }
 }
 
+#[derive(Debug, PartialEq)]
 /// RPC payload
-pub type RpcPayload<'a> = &'a [u8];
+pub struct RpcPayload<'a> {
+    /// Bytes
+    pub bytes: &'a [u8],
+}
+
+impl<'a> RpcPayload<'a> {
+    /// Length
+    pub fn length(&self) -> usize {
+        self.bytes.len()
+    }
+}
```

Comment:
This abstraction doesn't make sense and introduces addition noice by accessing the bytes field. Please impl at least Deref and DerefMut - this would also obsolete the `length` fn.

Reply:
mhh... 

1) kind of "Primitive Obsession" (http://wiki.c2.com/?PrimitiveObsession) imho.

`pub enum Message<'a> {
    RpcMessage(Header, RpcPayload<'a>),
    SdMessage(Header, SdPayload),
`
is providing some symmetry to the code, too!

2) what about future support of SOME/IP TP where an RPC message will contain only a chunk of a final "payload" with an additional index / offset.. => new Message Type or extend the RpcMessage struct?

Reply:
A slice is not a primitive. It provides functionality that you probably want for  a bytes slice ;-) 

Regarding 2. This would be a new enum item - otherwise you would break the API 🤷

Reply:
Ok!

Reply:
done.